test/userController.test.js

```diff
@@ -94,3 +94,143 @@ describe('GET /api/v1/users/:user_id endpoint', () => {
+      expect(album).toHaveProperty('year');
+      // Add more checks for album properties if needed
+    });
+
+    expect(response.body.user).not.toHaveProperty('password');
+  });
+
+  it('should return a 401 status if the user is not authenticated', async () => {
+    // Act: Attempt to get the current user's information with purchased albums without authentication
+    const response = await request(app).get(
+      '/api/v1/users/getCurrentUserWithPurchasedAlbums'
+    );
+
+    // Assert: Check the response status for 401
+    expect(response.status).toBe(401);
+  });
+});
+
+describe('GET /api/v1/users/showCurrentUser endpoint', () => {
+  it('should return the data of the currently authenticated user', async () => {
+    // Act: Log in and get a signed cookie
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Act: Get the current user's information using the authenticated cookie
+    const response = await request(app)
+      .get('/api/v1/users/showCurrentUser')
+      .set('Cookie', [signedCookie]);
+
+    // Assert: Check the response status and user information
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('user');
+    expect(response.body.user).toMatchObject({
+      name: 'ava',
+      email: 'ava@ava.com',
+      role: 'user',
+    });
+    expect(response.body.user).not.toHaveProperty('password');
+  });
+
+  it('should return a 401 status if the user is not authenticated', async () => {
+    // Act: Attempt to get the current user's information without authentication
+    const response = await request(app).get('/api/v1/users/showCurrentUser');
+
+    // Assert: Check the response status for 401
+    expect(response.status).toBe(401);
+  });
+});
+
+describe('GET /api/v1/users endpoint', () => {
+  it('should return a list of users without the password field', async () => {
+    // Act: Log in and get a signed cookie
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Act: Get a list of users using the authenticated cookie
+    const response = await request(app)
+      .get('/api/v1/users')
+      .set('Cookie', [signedCookie]);
+
+    // Assert: Check the response status and the list of users
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('users');
+    expect(Array.isArray(response.body.users)).toBe(true);
+
+    // Check that each user in the list doesn't have a password field
+    response.body.users.forEach((user) => {
+      expect(user).toHaveProperty('name');
+      expect(user).toHaveProperty('email');
+      expect(user).toHaveProperty('role');
+      expect(user).not.toHaveProperty('password');
+    });
+  });
+
+  it('should return a 401 status if the user is not authenticated', async () => {
+    // Act: Attempt to get a list of users without authentication
+    const response = await request(app).get('/api/v1/users');
+
+    expect(response.status).toBe(401);
+  });
```

Comment:
One additional test we can have is to verify that this endpoint only works for admin users and not regular user roles:

```suggestion
  });
  
  it('should return a 403 status if the user is not an admin', async () => {
    // Arrange: Log in as a regular user and get a signed cookie
    const signedCookie = await loginAndReturnCookie(testUserCredentials);

    // Act: Attempt to get a list of users using the authenticated cookie
    const response = await request(app).get('/api/v1/users').set('Cookie', [signedCookie]);

    // Assert: Check the response status for 403
    expect(response.status).toBe(403);
  });
```